servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -470,10 +474,10 @@ private boolean failIfWriteActive(ChannelOutboundListener newChannelOutboundList
             // It is possible that we have set the writeSubscriber, then the channel becomes inactive, and we will
             // never notify the write writeSubscriber of the inactive event. So if the channel is inactive we notify
             // the writeSubscriber.
-            // It is also possible that Channel is in closing state, we should abort new writes if a closeReason was
-            // observed:
+            // It is also possible that Channel is in closing state, we should abort new writes from the client-side
+            // if a closeReason was observed:
             CloseEvent closeReason = null;
-            if (!channel().isActive() || (closeReason = this.closeReason) != null) {
+            if (!channel().isActive() || (isClient && (closeReason = this.closeReason) != null)) {
```

Comment:
@NiteshKant upgrade to TLSv1.3 (client starts writing data earlier that server completes the handshake) helped to discover this issue, resulting in a lot of failed tests: https://ci.servicetalk.io/job/servicetalk-java11-prb/1521/
My fix isn't great: https://github.com/apple/servicetalk/pull/1141/commits/5c5b41c32b1f9dd3947bd44042f2f810e0698188. Open for suggestions.